lib/modules/manager/tekton/extract.ts

```diff
@@ -0,0 +1,97 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type { PackageDependency, PackageFile } from '../types';
+import type { TektonBundle, TektonResource } from './types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string
+): PackageFile | null {
+  logger.trace('tekton.extractPackageFile()');
+  const deps: PackageDependency[] = [];
+  let docs: any[];
+  try {
+    docs = loadAll(content) as TektonResource[];
+  } catch (err) {
+    logger.debug(
+      { err, fileName },
+      'Failed to parse YAML resource as a Tekton resource'
+    );
+    return null;
+  }
+  for (const doc of docs) {
```

Comment:
filter docs here, if there can be null / undefiend docs. I'm not sure if this can really happen
```suggestion
  for (const doc of docs.filter(is.truthy)) {
```

I'm also not sure if `loadAll` can return `null`, please check other managers, which use this function

Reply:
I moved this to `getDeps` so I could also completely remove the usage of `doc?` within it.